processing/src/main/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapter.java

```diff
@@ -234,13 +235,17 @@ public Metadata getMetadata()
       // Virtual columns cannot depend on each other, so we don't need to check transitive dependencies.
       if (baseColumns.containsAll(virtualColumn.requiredColumns())) {
```

Comment:
what happens if one of the required columns is a virtual column?

Reply:
Virtual columns currently can't use other virtual columns as input (they're each constructed independently from the base columns), if you tried I think it would just operate as if you referenced a non-existent column